src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1024,43 +1024,73 @@ ss::future<add_paritions_tx_reply> tx_gateway_frontend::do_add_partition_to_tx(
                 res_partition.error_code = tx_errc::none;
                 res_topic.results.push_back(res_partition);
             } else {
-                bfs.push_back(_rm_partition_frontend.local().begin_tx(
-                  ntp, tx.pid, tx.tx_seq, tx.timeout_ms, timeout));
+                new_partitions.push_back(ntp);
             }
         }
         response.results.push_back(res_topic);
     }
 
-    auto brs = co_await when_all_succeed(bfs.begin(), bfs.end());
-
     std::vector<tm_transaction::tx_partition> partitions;
-    for (auto& br : brs) {
-        auto topic_it = std::find_if(
-          response.results.begin(),
-          response.results.end(),
-          [&br](const auto& r) { return r.name == br.ntp.tp.topic(); });
-        vassert(
-          topic_it != response.results.end(),
-          "can't find expected topic {}",
-          br.ntp.tp.topic());
-        vassert(
-          std::none_of(
-            topic_it->results.begin(),
-            topic_it->results.end(),
-            [&br](const auto& r) {
-                return r.partition_index == br.ntp.tp.partition();
-            }),
-          "partition {} is already part of the response",
-          br.ntp.tp.partition());
-        if (br.ec == tx_errc::none) {
-            partitions.push_back(
-              tm_transaction::tx_partition{.ntp = br.ntp, .etag = br.etag});
+    std::vector<begin_tx_reply> brs;
+    auto retries = _metadata_dissemination_retries;
+    auto delay_ms = _metadata_dissemination_retry_delay_ms;
```

Comment:
Why is this related with metadata dissemination ?

Reply:
we need to retry `begin_tx` because reconfiguration, partition movement or re-election may cause metadata become stale